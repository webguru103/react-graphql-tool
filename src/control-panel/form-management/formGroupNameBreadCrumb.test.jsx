import React from 'react';
import {
  render, cleanup, waitForElement,
} from 'react-testing-library';
import { withContext } from '../../../tests/test-helpers/contextWrapper';
import FormGroupNameBreadCrumb from './formGroupNameBreadCrumb';

describe('Form Group Name Bread Crumb', () => {
  afterEach(cleanup);
  const Component = withContext(FormGroupNameBreadCrumb);
  const mockForm = {
    id: 1,
    formName: 'mockForm',
    formStatus: 'ACTIVE',
    formGroupId: 3,
  };
  const mockFormGroup = {
    id: 1,
    formGroupName: 'mockName',
  };
  test('renders', () => {
    const { container } = render(<Component classes={{}} form={mockForm} formGroups={[mockFormGroup]} />);
    expect(container).toMatchSnapshot();
  });

  test('should have formGroupName in breadcrumb', () => {
    const { container } = render(<Component classes={{}} form={mockForm} formGroups={[mockFormGroup]} />);
    const breadCrumb = waitForElement(() => container(mockFormGroup.formGroupName));
    expect(breadCrumb).toBeDefined();
  });
});
