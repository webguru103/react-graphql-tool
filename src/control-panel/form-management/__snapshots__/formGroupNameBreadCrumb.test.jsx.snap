// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Group Name Bread Crumb renders 1`] = `
<div>
  <span />
</div>
`;
